src/index.js

```diff
@@ -1,25 +1,34 @@
-import { useState, useEffect } from "react";
+import React, { useState, useRef, useEffect } from "react";
 
 const noop = () => undefined;
 export default function useInputDebounce(
   effect = noop,
-  { initial, delay = 0, minLength = 0 }
+  { initial = "", delay = 0, minLength = 0 }
 ) {
   const [value, setValue] = useState(initial);
+  const debounceEnabled = useRef(true);
   const [onChange] = useState(() => e => {
-    const value = e && e.target ? e.target.value : e;
-    setValue(value || "");
+    const value = e && e.target ? e.target.value : undefined;
+    if (value) setValue(value);
   });
 
+  const updateValue = value => {
+    debounceEnabled.current = false;
+    setValue(String(value));
+  };
+
   useEffect(() => {
     let timeout;
-    // Avoid to call effect when it first loads unless you do not
-    // specify an inital value
-    if (value !== undefined && value.length >= minLength) {
-      timeout = setTimeout(() => effect(value, onChange), delay);
+    if (
+      value !== undefined &&
+      value.length >= minLength &&
+      debounceEnabled.current
+    ) {
+      timeout = setTimeout(() => effect(value, updateValue), delay);
     }
+    debounceEnabled.current = true;
     return () => clearTimeout(timeout);
   }, [value, delay]);
 
-  return { value: value || "", onChange };
+  return { value, onChange };
```

Comment:
For the comment left at line number 6 we should keep ``` { value: value || "", onChange }; ``` because  when we render the input the first time when we have not type anything yet, we do not want to get called the effect function.

Reply:
I added this back in, though it doesn't appear to run the effect without it?  Doesn't hurt through.